crates/zizmor/src/audit/impostor_commit.rs

```diff
@@ -120,13 +120,11 @@ impl Audit for ImpostorCommit {
             )));
         }
 
-        let Some(client) = state.github_client() else {
-            return Err(AuditLoadError::Skip(anyhow!(
-                "can't run without a GitHub API token"
-            )));
-        };
-
-        Ok(ImpostorCommit { client })
+        state
+            .github_client()
```

Comment:
Flagging: maybe not worth fixing with this PR, but it seems non-ideal that we now have to re-check the error state on `GitHubClient` on each audit initialization. Maybe we should create a single `Option<GitHubClient>` on the audit state instead.

(Feel free to look into that if you feel like it; otherwise that can wait until another refactor 🙂)

Reply:
No worries at all, thanks for poking at it!